saleor/product/utils/variant_prices.py

```diff
@@ -5,52 +5,63 @@
 from django.db.models.query_utils import Q
 from prices import Money
 
-from ...discount.utils import fetch_active_discounts
-from ..models import Product
+from ...discount.utils import calculate_discounted_price, fetch_active_discounts
+from ..models import Product, ProductChannelListing, ProductVariantChannelListing
 
 
-def _get_product_minimal_variant_price(product, discounts) -> Optional[Money]:
+def _get_product_discounted_price(
+    product_channel_listing, discounts
+) -> Optional[Money]:
     # Start with the product's price as the minimal one
     minimal_variant_price = None
-    for variant in product.variants.all():
-        variant_price = variant.get_price(discounts=discounts)
+    variants_channel_listing = ProductVariantChannelListing.objects.filter(
+        variant__product_id=product_channel_listing.product_id,
+        channel_id=product_channel_listing.channel_id,
+    )
+    product = product_channel_listing.product
+    collections = (product.collections.all(),)
+    for variant_channel_listing in variants_channel_listing:
+        variant_price = calculate_discounted_price(
+            product=product,
+            price=variant_channel_listing.price,
+            collections=collections,
+            discounts=discounts,
+        )
```

Comment:
Are you sure this works correctly? I am curious why you creating a tuple from collections. As I see `calculate_discounted_price` require `Iterable` and then collections are provided as an argument to `get_product_discounts`, where collections are used in this way:
`product_collections = set(pc.id for pc in collections)`. 

Maybe it should be like:
`collections = list(product.collections.all())` ?

Reply:
Good catch, thanks. I will fix it.